### ISSUE TITLE
arcs on dselector are too thin

### ISSUE THREAD

**Issue by decentjohn** [decentespresso/de1app#246]
The left and right arcs are slightly too thin:
![Screen Shot 2022-06-02 at 4 06 55 PM](https://user-images.githubusercontent.com/11965250/171661389-0c305bcc-aa71-4f21-bc48-e342213a35e7.JPG)

I think this is simply how Tk draws them, compared to vertical/horizontal lines.

One fix is to add +1 to the width of the arcs:

```
			if { $radius1 > 0 } {
				lappend ids [$can create arc $x0 [expr {$y0+$radius1+1.0}] [expr {$x0+$radius1+1.0}] $y0 -style arc -outline $colour \
					-width [expr {$arc_width+1}] -tags [list ${main_tag}-nw {*}$tags] -start 90 -disabledoutline $disabled -state "hidden"]
			}
			if { $radius2 > 0 } {
				lappend ids [$can create arc [expr {$x1-$radius2-1}] $y0 $x1 [expr {$y0+$radius2+1}] -style arc -outline $colour \
					-width [expr {$arc_width+1}] -tags [list ${main_tag}-ne {*}$tags] -start 0 -disabledoutline $disabled -state "hidden"]
			}
			if { $radius3 > 0 } {
				lappend ids [$can create arc [expr {$x1-$radius3-1.0}] $y1 $x1 [expr {$y1-$radius3-1.0}] -style arc -outline $colour \
					-width [expr {$arc_width+1}] -tags [list ${main_tag}-se {*}$tags] -start -90 -disabledoutline $disabled -state "hidden"]
			}			
			if { $radius4 > 0 } {
				lappend ids [$can create arc $x0 [expr {$y1-$radius4-1.0}] [expr {$x0+$radius4+1.0}] $y1 -style arc -outline $colour \
					-width [expr {$arc_width+1}] -tags [list ${main_tag}-sw {*}$tags] -start 180 -disabledoutline $disabled -state "hidden"]
			}

```

You can see how this looks if I change just one part of the arc:
![Screen Shot 2022-06-02 at 4 03 00 PM](https://user-images.githubusercontent.com/11965250/171661707-bc0f9928-f61d-42b3-9119-6d2dad9cd6f1.JPG)

doing it to the 4 arcs fixes the appearance:

![Screen Shot 2022-06-02 at 4 09 20 PM](https://user-images.githubusercontent.com/11965250/171661541-06b150fd-9c72-4304-bed0-01bff2e1d98f.JPG)

I noticed that even when I drew with a thicker -width, the arcs were still off by one:
![Screen Shot 2022-06-02 at 4 05 21 PM](https://user-images.githubusercontent.com/11965250/171661822-f5ff53d2-b76b-486b-a7eb-62f80163088f.JPG)

I'm not sure if this is how you want this fixed, or prefer another method?

**Comment by ebengoechea** [comment]
@decentjohn Are the captures from tablet or PC/Mac?

The corners are coded to have one pixel less than the sides, intentionally, because this is how they looked best when using thin borders *on my Teclast tablet*. Having them the same width showed them too wide, not matching the sides lines. But this was different on my PC (which was better when using the same width for sides and corners), so I prioritized the tablet case.

It is my suspicion also that this may depend on the specific width values being even or odd in the target resolution, as they are specified in a bigger resolution and then rescaled. So, rounding may produce different results in different resolutions.

**Comment by decentjohn** [comment]
Yeah, this might be an artifact of anti-aliasing and might look different at other resolutions.

Here is a picture from my Teclast:
![Screenshot_20220602-172514](https://user-images.githubusercontent.com/11965250/171678266-01c0fc75-c198-4131-ad58-547774db71e0.png)

Note that I'm not specifying -width -- the default is active.

I will try to double resolution and see.

**Comment by decentjohn** [comment]
At 2560x1600 it's still looking good, though there is a "darkness" to the arcs that isn't on the lines.  I'm assuming that's a downscaling artifact caused by the antialiasing.

![Screen Shot 2022-06-02 at 5 27 29 PM](https://user-images.githubusercontent.com/11965250/171678710-e9dd260e-9cfe-47cd-b151-379218fa75b4.JPG)
![Screen Shot 2022-06-02 at 5 27 20 PM](https://user-images.githubusercontent.com/11965250/171678715-9305c1d5-f01b-41b6-9a7f-ed5884f5699a.JPG)

**Comment by ebengoechea** [comment]
@decentjohn 
Is the change on nightly? It would be easier if you just comment the following lines (starting in line 6252) in ```proc rounded_rectangle_outline```:
```
if { $width > 1 } {
    # Adjustment to look better under Android, that uses dithering
    set arc_width [expr {$width-1}]
} else {
    set arc_width $width
}
```

**Comment by decentjohn** [comment]
I haven't committed my change, as I was waiting until we settled our conversation.  I will switch to your method for this.

**Comment by decentjohn** [comment]
![Screen Shot 2022-06-08 at 7 53 10 AM](https://user-images.githubusercontent.com/11965250/172542045-3c36eeae-ad94-4fde-b22a-2dcde71daadf.JPG)

Done, and I switched to -width 2 for my dselectors, as the color change that dithering causes, goes away then.  This looks nice.

**Comment by ebengoechea** [comment]
@decentjohn Is that capture from your tablet?

**Comment by decentjohn** [comment]
this is from my teclast tablet, thickness looks good, darkness of color slightly different. I'm guessing that the "fix" would be to make the straight lines drawn with dithering.  Do you know how to do that?

![Screenshot_20220608-070930](https://user-images.githubusercontent.com/11965250/172544103-feeb2a91-952c-4188-b8ab-5c9080a914b1.png)

**Comment by ebengoechea** [comment]
Not possible as far as I know. It seems the problems arise from different dithering behaviour on lines and arcs and between platforms (PC/Mac/Android), but it is automatic, there's no argument for controlling it on canvas elements.

**Comment by ebengoechea** [comment]
This is how it looks in my tablet when using ```-width 3``` (and commenting the code  that removes 1 px from the arcs widths). It looks as if the arc width is one pixel wider than the lines, likely as a consequence of the dithering, and it's the reason I removed one pixel, as it makes the widths of the lines and arcs match better. This does not happen when the width is 1 px, and does not happen on PC neither (irrespective of the width value).
![DUI_dbutton_outline_width3px](https://user-images.githubusercontent.com/60866/172554024-4266163b-1c50-439a-be19-4ce96a422f6d.png)
